automation/taskcluster/lib/artifacts.py

```diff
@@ -0,0 +1,22 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import re
+import subprocess
+
+
+def from_gradle():
+    process = subprocess.Popen(["./gradlew", "--no-daemon", "printModules"], stdout=subprocess.PIPE)
+    (output, err) = process.communicate()
+    exit_code = process.wait()
+
+    if exit_code is not 0:
+        print "Gradle command returned error:", exit_code
+
+    tuples = re.findall('module: name=(.*) buildPath=(.*)', output, re.M)
+    return map(lambda (name, build_path): {
+        'name': name[1:],
+        'artifact': 'public/build/' + name[1:].replace('-', '.', 1) + '.maven.zip',
+        'path': build_path + '/target.maven.zip'
+    }, tuples)
```

Comment:
Note that artifacts.yml is smaller than all gradle modules. All gradle modules also contains sample apps and modules we do not publish (e.g. tooling-lint). As far as I can see we do not filter those out yet. There may be no good automated way to figure out what module should be published - so I wonder whether we should just add an additional flag to settings.gradle, where we configure other things already like the description.

Reply:
I had this in mind too last night but wasn't 100% sure it's accurate so I'm happy you raised it. I think the solutions with additional flags in `settings.gradle` is great as it scales better if we wanted to add more tasks in the graph (signing, checksums, etc) later on.

Reply:
Really great point! I should do a more-complete diff of tasks created before vs. after this change.
Great catch finding this in code review, thanks! :100: